modules/core/src/shaderlib/lighting/lighting.glsl.js

```diff
@@ -19,51 +19,94 @@
 // THE SOFTWARE.
 
 // lighting
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
 
-export default `\
-#define MAX_NUM_OF_LIGHTS 5
+// lighting
 
-// TODO these should be using lighting_ prefix
-uniform vec3 lighting_lightPositions[MAX_NUM_OF_LIGHTS];
-uniform vec2 lighting_lightStrengths[MAX_NUM_OF_LIGHTS];
-uniform float lighting_ambientRatio;
-uniform float lighting_diffuseRatio;
-uniform float lighting_specularRatio;
-uniform int lighting_numberOfLights;
+export default `\
+#define MAX_LIGHTS 5
 
-float lighting_getLightWeight(vec3 position_worldspace_vec3, vec3 normals_worldspace) {
-  float lightWeight = 0.0;
+struct AmbientLight {
+ vec3 color;
+ float intensity;
+};
 
-  vec3 normals_worldspace_vec3 = normals_worldspace.xyz;
+struct PointLight {
+ vec3 color;
+ float intensity;
+ vec3 position;
+};
 
-  vec3 camera_pos_worldspace = project_uCameraPosition;
-  vec3 view_direction = normalize(camera_pos_worldspace - position_worldspace_vec3);
+struct DirectionalLight {
+  vec3 color;
+  float intensity;
+  vec3 direction;
+};
+ 
+uniform AmbientLight lighting_ambientLight;
+uniform PointLight lighting_pointLight[MAX_LIGHTS];
+uniform DirectionalLight lighting_directionalLight[MAX_LIGHTS];
+uniform int lighting_pointLightNumber;
```

Comment:
Nit: `lighting_pointLightNumber` -> `lighting_pointLighCount` ? for directional light too.